Lib/test/test_types.py

```diff
@@ -1360,6 +1360,24 @@ class C: pass
         D = types.new_class('D', (A(), C, B()), {})
         self.assertEqual(D.__bases__, (A1, A2, A3, C, B1, B2))
 
+    def test_get_original_bases(self):
+        T = typing.TypeVar('T')
+        class A: pass
+        class B(typing.Generic[T]): pass
+        class C(B[int]): pass
+        class D(B[str], float): pass
+        self.assertIsNone(types.get_original_bases(A))
+        self.assertEqual(types.get_original_bases(B), (typing.Generic[T],))
+        self.assertEqual(types.get_original_bases(C), (B[int],))
+        self.assertIsNone(types.get_original_bases(int))
+        self.assertEqual(types.get_original_bases(D), (B[str], float))
+
+        class E(list[T]): pass
+        class F(list[int]): pass
+
+        self.assertEqual(types.get_original_bases(E), (list[T],))
+        self.assertEqual(types.get_original_bases(F), (list[int],))
+
```

Comment:
Could you add a test for a class inheriting from `typing.NamedTuple`, since that's something of a special case? `typing.NamedTuple` is a function, but it's acceptable as a "base class" due to the fact that `__mro_entries__` is monkey-patched onto the function here:

https://github.com/python/cpython/blob/4dc339b4d69195448207e1faecc3e258700daf33/Lib/typing.py#L2967

```suggestion

class G(typing.NamedTuple):
    x: int

self.assertIs(types.get_original_bases(G)[0], typing.NamedTuple)

```